src/lib.rs

```diff
@@ -0,0 +1,886 @@
+use lightning::ln::channelmanager;
+use lightning::ln::channelmanager::{
+	ChainParameters, ChannelManagerReadArgs, SimpleArcChannelManager,
+};
+use lightning::ln::peer_handler::{IgnoringMessageHandler, MessageHandler, SimpleArcPeerManager};
+use lightning::ln::{PaymentHash, PaymentPreimage, PaymentSecret};
+use lightning::routing::gossip;
+use lightning::routing::gossip::P2PGossipSync;
+use lightning::routing::scoring::ProbabilisticScorer;
+
+use lightning::util::config::UserConfig;
+use lightning::util::ser::ReadableArgs;
+
+use lightning_background_processor::BackgroundProcessor;
+use lightning_background_processor::GossipSync as BPGossipSync;
+use lightning_persister::FilesystemPersister;
+
+use lightning_net_tokio::SocketDescriptor;
+
+use lightning_invoice::utils::DefaultRouter;
+use lightning_invoice::{payment, Currency, Invoice};
+
+use bdk::blockchain::esplora::EsploraBlockchain;
+use bdk::blockchain::{GetBlockHash, GetHeight};
+use bdk::database::MemoryDatabase;
+
+use bitcoin::hashes::sha256::Hash as Sha256;
+use bitcoin::hashes::Hash;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::BlockHash;
+
+use rand::RngCore;
+
+use std::collections::HashMap;
+use std::fs;
+use std::net::SocketAddr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+use std::time::{Duration, SystemTime};
+
+// TODO: Is MemoryDatabase okay to use?
+
+const CHANNEL_BUF_SIZE: usize = 1000;
+
+#[derive(Debug, Clone)]
+/// Represents the configuration of an `LdkLite` instance.
+pub struct LdkLiteConfig {
+	/// The path where the underlying LDK and BDK persist their data.
+	pub storage_dir_path: String,
+	/// The URL of the utilized Esplora server.
+	pub esplora_server_url: String,
+	/// The used Bitcoin network.
+	pub network: bitcoin::Network,
+	/// The TCP port the network node will listen on.
+	pub listening_port: u16,
+	/// The default CLTV expiry delta to be used for payments.
+	pub default_cltv_expiry_delta: u32,
+}
+
+/// A builder for an [`LdkLite`] instance, allowing to set some configuration and module choices from
+/// the getgo.
+#[derive(Debug, Clone)]
+pub struct LdkLiteBuilder {
+	config: LdkLiteConfig,
+}
+
+impl LdkLiteBuilder {
+	/// Creates a new builder instance with the default configuration.
+	pub fn new() -> Self {
+		// Set the config defaults
+		let storage_dir_path = "/tmp/ldk_lite/".to_string();
+		let esplora_server_url = "https://blockstream.info/api".to_string();
+		let network = bitcoin::Network::Testnet;
+		let listening_port = 9735;
+		let default_cltv_expiry_delta = 144;
+
+		let config = LdkLiteConfig {
+			storage_dir_path,
+			esplora_server_url,
+			network,
+			listening_port,
+			default_cltv_expiry_delta,
+		};
+
+		Self { config }
+	}
+
+	/// Creates a new builder instance from an [`LdkLiteConfig`].
+	pub fn from_config(config: LdkLiteConfig) -> Self {
+		Self { config }
```

Comment:
set config ?
And this method should be named just config, so that we can use it like builder().config(liteConfig).build()

Reply:
No, in this case the builder consumes the config, which is a conversion. Typically this is indicated by the `from_` prefix, which is also in line with [rust naming guidelines](https://rust-lang.github.io/api-guidelines/naming.html). Also, `config` would be reserved for the getter, not the setter.